### PR TITLE
Coverage

### PR DESCRIPTION
**Pull request by marshall-mcmullen**

### CODE REVIEW

File: bin/etest [hunk 1]
```diff
@@ -243,10 +182,93 @@ global_setup()
...
+
+        # If we can't use elogfile and we don't want verbose output we are forced to do some trickery to get the desired
+        # result. Yes this is a bit convoluted since we need to basically show the terse output on the console but also
+        # send a copy of both stdout and stderr to the log file. So basically we send stdout to the logfile, then we
+        # create a new file descriptor (3) and associate that with what is our current stderr. Then we can send stderr to
+        # where stdout has been redirect so that they both end up in the logfile.
+        exec 1>${ETEST_LOG}
+        exec 3>&2
+        exec 2>&1
+        exec {ETEST_OUT_FD}<&3
```
Comment (ericpe) [inline bin/etest:252]: image

**Review by mindkeep — APPROVED** [review]
LGTM! (aside from @ericpe 's comment). Hi Marshal.

**Comment by marshall-mcmullen** [comment]
> LGTM! (aside from @ericpe 's comment). Hi Marshal.

Howdy Dave! Good to hear from you!! I need to finish this off when I get some more spare time. Gonna be pretty slick.

File: Makefile [hunk 2]
```diff
@@ -113,6 +113,7 @@ DISTROS =			\
 	fedora:33		\
 	fedora:32		\
 	gentoo/stage3	\
```
Comment (jwkblades) [inline Makefile:115]: Nit: Spacing is off here

File: bin/etest [hunk 3]
```diff
@@ -56,6 +57,10 @@ $(opt_parse \
     "+verbose v=${VERBOSE:-0}  | Verbose output." \
     ":work_dir                 | Temporary location where etest can place temporary files.  This location will be both
                                  created and deleted by etest." \
+    "+coverage                 | Run etest through kcov coverage tool to identify and report code coverage for all
+                                 executed code." \
```
Comment (jwkblades) [inline bin/etest:61]: This is going to be amazing!

**Review by jwkblades — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (marshall-mcmullen) [reply Makefile:115]: Thanks for catching that!

File: bin/etest (hunk 3 above)
Comment (marshall-mcmullen) [reply bin/etest:61]: yes, very much agree. Early protoypes of this look amazeballs. I put this on hold as I had more urgent things I had to finish, but I'm going to come back to this soon. BTW, speaking of **AMAZING** check this out:

https://elibs.github.io/ebash

**Pull request closed without merging**